config_parser.c

```diff
@@ -0,0 +1,180 @@
+/*
+ *  This file is a part of OpenVPN-GUI -- A Windows GUI for OpenVPN.
+ *
+ *  Copyright (C) 2016 Selva Nair <selva.nair@gmail.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program (see the file COPYING included with this
+ *  distribution); if not, write to the Free Software Foundation, Inc.,
+ *  59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <windows.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <wchar.h>
+#include "main.h"
+#include "config_parser.h"
+
+static int
+legal_escape(wchar_t c)
+{
+    wchar_t *escapes = L"\"\' \\"; /* space, ", ' or backslash */
+    return (wcschr(escapes, c) != NULL);
+}
+
+static int
+copy_token(wchar_t **dest, wchar_t **src, wchar_t* delim)
+{
+    wchar_t *p = *src;
+    wchar_t *s = *dest;
+
+    /* copy src to dest until delim character with escaped chars converted */
+    for ( ; *p != L'\0' && wcschr(delim, *p) == NULL; p++, s++)
+    {
+        if (*p == L'\\' && legal_escape(*(p+1)))
+        {
+            *s = *(++p);
+        }
+        else if (*p == L'\\')
+        {
+            return -1; /* parse error -- illegal backslash in input */
+        }
+        else
+            *s = *p;
+    }
+    /* at this point p is one of the delimiters or null */
+    *s = L'\0';
+    *src = p;
+    *dest = s;
+    return 0;
+}
+
+static int
+tokenize(config_entry_t *ce)
+{
+    wchar_t *p, *s;
+    p = ce->line;
+    s = ce->sline;
+    unsigned int i = 0;
+    int status;
+
+    for ( ; *p != L'\0';  p++, s++)
+    {
+        if (*p == L' ' || *p == L'\t') continue;
+
+        if (_countof(ce->tokens) <= i)
```

Comment:
That occurred to me as well. I did not want to ignore comments, but we can use a single token per comment (whole line or trailing part). Will do.
That said, the present code will just truncate the comment content, so no harm done as we are not (yet) using the parsed results to re-write the ovpn file or anything like that. So this could be done as a followup?